launcher/minecraft/launch/AutoInstallJava.cpp

```diff
@@ -0,0 +1,233 @@
+                        emit logLine(tr("The automatic Java mechanism detected an x32 java."), MessageLevel::Info);
+                    }
+                    setJavaPath(java->path);
+                    return;
+                }
+            }
+            emit logLine(tr("No compatible java version was found. Using the default one."), MessageLevel::Warning);
+            emitSucceeded();
+        });
+        connect(m_current_task.get(), &Task::progress, this, &AutoInstallJava::setProgress);
+        connect(m_current_task.get(), &Task::stepProgress, this, &AutoInstallJava::propagateStepProgress);
+        connect(m_current_task.get(), &Task::status, this, &AutoInstallJava::setStatus);
+        connect(m_current_task.get(), &Task::details, this, &AutoInstallJava::setDetails);
+        emit progressReportingRequest();
+        return;
+    }
+    if (m_supported_arch.isEmpty()) {
+        emit logLine(tr("Your system(%1 %2) is not compatible with auto java download. Using the default java path.")
+                         .arg(SysInfo::currentSystem(), SysInfo::useQTForArch()),
+                     MessageLevel::Warning);
+        emitSucceeded();
+        return;
+    }
+    auto wantedJavaName = packProfile->getProfile()->getCompatibleJavaName();
+    if (wantedJavaName.isEmpty()) {
+        emit logLine(tr("Your meta informtation is old or doesn't have the information necesary to determine what java should be used. "
+                        "Using the default java path."),
+                     MessageLevel::Warning);
+        emitSucceeded();
+        return;
+    }
+    QDir javaDir(APPLICATION->javaPath());
+    auto wantedJavaPath = javaDir.absoluteFilePath(wantedJavaName);
+    if (QFileInfo::exists(wantedJavaPath)) {
+        setJavaPathFromPartial();
+        return;
+    }
+    auto versionList = APPLICATION->metadataIndex()->get("net.minecraft.java");
+    m_current_task = versionList->getLoadTask();
+    connect(m_current_task.get(), &Task::succeeded, this, &AutoInstallJava::tryNextMajorJava);
+    connect(m_current_task.get(), &Task::failed, this, &AutoInstallJava::emitFailed);
+    connect(m_current_task.get(), &Task::progress, this, &AutoInstallJava::setProgress);
+    connect(m_current_task.get(), &Task::stepProgress, this, &AutoInstallJava::propagateStepProgress);
+    connect(m_current_task.get(), &Task::status, this, &AutoInstallJava::setStatus);
+    connect(m_current_task.get(), &Task::details, this, &AutoInstallJava::setDetails);
+    emit progressReportingRequest();
+}
+
+void AutoInstallJava::setJavaPath(QString path)
+{
+    auto settings = m_instance->settings();
+    settings->set("OverrideJavaLocation", true);
+    settings->set("JavaPath", path);
+    emit logLine(tr("Compatible java found at: %1.").arg(path), MessageLevel::Info);
+    emitSucceeded();
+}
+
+void AutoInstallJava::setJavaPathFromPartial()
+{
+    QString executable = "java";
+#if defined(Q_OS_WIN32)
+    executable += "w.exe";
+#endif
```

Comment:
this is copy-pasted from JavaUtils::getPrismJavaBundle. it'd be nice if we re-use it instead of doing this each time